tools/bin/ariaExtract.py

```diff
@@ -142,6 +142,34 @@ def main():
     logging.basicConfig(level=log_level, format=ARIAtools.util.log.FORMAT)
     LOGGER.info('Extract Product Function')
 
+    # Check whether all necessary inputs were specified.
+    # some products require a DEM to extract -- if any of those are requested,
+    # ensure that a valid DEM is specified
+    if args.layers:
+        # format list of layers
+        layers = list(args.layers.split(','))
+        layers = [i.replace(' ', '') for i in layers]
+        layers = ['all' if layer.lower() == 'all'
+                  else layer for layer in layers]
+
+        # list of layers requiring DEM for extraction
+        layers_requiring_dem = ['all',
+                                'bPerpendicular',
+                                'bParallel',
+                                'incidenceAngle',
+                                'lookAngle',
+                                'azimuthAngle',
+                                'solidEarthTide']
+
+        # check that DEM is specified depending on layers requested
+        intersecting_layers = [layer for layer in layers
+                               if layer in layers_requiring_dem]
+        if intersecting_layers != []:
+            assert args.demfile is not None, (
```

Comment:
Should this be an assertion, or should we logically and the assertion with the conditional on the line above then raise an Exception and issue a `LOGGER.error` message here if passes conditional? IMO this would be more simple to understand but same behavior